packages/material-ui-lab/src/Alert/Alert.d.ts

```diff
@@ -1,6 +1,7 @@
 import * as React from 'react';
 import { StandardProps } from '@material-ui/core';
 import { PaperProps } from '@material-ui/core/Paper';
+import '../theme'; // augment types for MUI theme
```

Comment:
I wouldn't do this at that point just to not confuse this with runtime side-effects. Let's stick with explicit module augmentation.

Reply:
I can see where this would be confusing for devs working on lab, that's why I added a comment on the import.

Are you suggesting we force users to apply the module augmentation themselves? Why would we do that if we could do it for them here? @oliviertassinari and I both agree that handling it automatically is better.

Reply:
(For the record, I would rather trust @eps1lon than my own judgment on this topic 😅).

Reply:
Reading again I feel like I came off a little strong 😅  
Just would like to understand why we wouldn't do this for users / what the down side is.

If we still decide against this, then having the one line import for users is still a great improvement.
```ts
import '@material-ui/lab/theme';
```

Reply:
Maybe we could add semantic in the path of the import:

```
import '@material-ui/lab/themeTypeAugmentation';
```
?

Reply:
>  I both agree that handling it automatically is better.

I don't think it is applied automatically. I can check out this PR in a new project to see if it works. Maybe I'm missing something.

Since there are no docs included this should "just" work, right? As in I can install `core` and `lab` and then do

```ts
createMuiTheme({
  props: {
    MuiRating: {
      emptyLabelText: 'Not rated',
    },
  },
});
```
?

Reply:
@eps1lon Yeah it should apply automatically as long as there's an import for it somewhere in user src. I copied these changes into my app and I can immediately get TS passing for lab props and overrides (since I already have a lab component in use in my app). To test, as soon as I remove the imports in each lab component file then TS gives error.

Reply:
> Yeah it should apply automatically as long as there's an import for it somewhere in user src.

Which is a problem because you have to import first before you get the props in the theme typed. This is not how /core works. I'd rather not have this implicit magic and keep the modules as side-effect free as possible. Even if it's "just" type side-effects. 

This should be explicitly imported when using the theme e.g. `import '@material-ui/lab/theme-types'`.

Reply:
That's true that it's different than _core_. It seems like for this case you'd always import the component(s) first then see what defaults/tweaks need to be made in theme. But I hear you that this can feel magical.